app/customer.py

```diff
@@ -0,0 +1,47 @@
+from app.car import Car
+
+
+class Customer:
+    def __init__(self,
+                 name: str,
+                 product_cart: dict[str, int],
+                 location: list[int],
+                 money: int,
+                 car: Car) -> None:
+
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = car
+
+        self.__home_location = location
+        self.__money_to_spend_in_shop: dict[str, float] = {}
+
+    def __repr__(self) -> str:
+        return f"{self.name}"
+
+    def go_to_home(self) -> float:
+        if self.location == self.__home_location:
+            raise RuntimeError(f"Customer {self.name} is already at home!")
+
+        return self.go_to_location(self.__home_location)
+
+    def go_to_location(self, location: list[int]) -> float:
+        distance = calculate_distance(self.location, location)
+        self.location = location
+        return distance
+
+    def buy_product(self, product_price: float) -> None:
+        if self.money > product_price:
```

Comment:
The condition should check if `self.money` is greater than or equal to `product_price` to ensure the customer has enough funds to make the purchase.